plugin/src/main/java/org/opensearch/ml/action/tasks/GetTaskTransportAction.java

```diff
@@ -84,6 +141,111 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLTask
             log.error("Failed to get ML task " + taskId, e);
             actionListener.onFailure(e);
         }
+    }
+
+    private void processRemoteBatchPrediction(MLTask mlTask, String taskId, ActionListener<MLTaskGetResponse> actionListener) {
+
+        Map<String, Object> transformJob = mlTask.getTransformJob();
 
+        Map<String, String> parameters = new HashMap<>();
+        for (Map.Entry<String, ?> entry : transformJob.entrySet()) {
+            if (entry.getValue() instanceof String) {
+                parameters.put(entry.getKey(), (String) entry.getValue());
+            } else {
+                log.debug("Value for key " + entry.getKey() + " is not a String");
+            }
+        }
+
+        if (parameters.containsKey("TransformJobArn") && parameters.get("TransformJobArn") != null) {
+            String jobArn = parameters.get("TransformJobArn");
+            String transformJobName = jobArn.substring(jobArn.lastIndexOf("/") + 1);
+            parameters.put("TransformJobName", transformJobName);
+            parameters.remove("TransformJobArn");
+        }
+
+        RemoteInferenceInputDataSet inferenceInputDataSet = new RemoteInferenceInputDataSet(parameters, ActionType.BATCH_STATUS);
+        MLInput mlInput = MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inferenceInputDataSet).build();
+        String modelId = mlTask.getModelId();
+
+        try {
+            mlModelManager.getModel(modelId, null, null, ActionListener.wrap(model -> {
+                if (clusterService.state().metadata().hasIndex(ML_CONNECTOR_INDEX)) {
+                    ActionListener<Connector> listener = ActionListener.wrap(connector -> {
+                        connector.decrypt(BATCH_STATUS.name(), (credential) -> encryptor.decrypt(credential));
+                        RemoteConnectorExecutor connectorExecutor = MLEngineClassLoader
```

Comment:
Is connector access controller included? If it's just one line change, I think we should still add this access check.

Reply:
You mean to check the user's access to the connector

Reply:
We should then also add model access controller I guess